client/src/components/UserAvatar.jsx

```diff
@@ -0,0 +1,38 @@
+import { HiUserCircle, HiPencil } from 'react-icons/hi';
+import PropTypes from 'prop-types';
+
+
+export default function UserAvatar({imageRef, onUploadClick, onEditClick }) {
+  return (
+    <div className="relative group">
+      {imageRef?.current.src ? (
```

Comment:
what's the logic behind this?

Reply:
I've already created the ```imageRef= useRef(new Image())``` bcz if the profile image is already then just updated the src link, so here just to check the src is null or not.